addons/udes_stock/models/mixin_stock_model.py

```diff
@@ -40,6 +40,10 @@ def get_or_create(self, identifier, create=False, aux_domain=None):
         :kwargs:
             - create: Boolean
                 If true, and MSM_CREATE is true, a new object is created if needed
+            - aux_domain: list
+                An additional domain to add to the search
+            - no_results: Boolean
```

Comment:
Nit: Believe the name no_results can be changed as it is not understand easily that when no_results is False to raise error, how about smth like no_results_exception, you may need to change the logic in if condition and where is called.
if no_results_exception:
    raise ValidationError...

If you want by default to raise validation error may change the default to True

Reply:
I will ask Miquel, I worked this out with him. We took the logic from another function that does the same thing. I think that he will reply that we want to keep the logic of this function this way most probably. Changing the name is probably not an issue. Will update you here later.